src/platform/forms-system/src/js/utilities/data/reduceErrors.js

```diff
@@ -1,8 +1,90 @@
-// min/max length or item errors may show up as duplicates
-const errorExists = (list, name, index) =>
-  list.some(obj => obj.name === name && obj.index === index);
+import numberToWords from './numberToWords';
+
+// Process JSON-schema error messages for viewing
+
+/**
+ * Convert array zero-based number `test[0]` into a word `first test`
+ * @param {string} _ - Entire regex matching group (unused)
+ * @param {string} word - Word portion of the regexp that matches the array name,
+ *   e.g. in "array[0]", "array" would be the word
+ * @param {string} number - Number portion of the regexp that matches the number
+ * @returns {string} "{converted word number} {word}"
+ */
+export const replaceNumberWithWord = (_, word, number) => {
+  const num = parseInt(number, 10);
+  return `${
+    isNaN(num) || !isFinite(num) ? number : numberToWords(num + 1)
+  } ${word}`;
+};
+
+/**
+ * @typedef FormConfig~reviewErrorMessage
+ * @type {string|function}
+ * @param {number} index - provided to the function if the schema property is an
+ *  array. This index indicates the index of the entry with an error
+ * @returns {string} predefined error message
```

Comment:
I love all these JSDocs. So, so helpful. :bow:
Not as useful as TypeScript, of course, but it's the next best thing. :smile: